File: json_host/generate.py
import json
import os
from pathlib import Path

INPUT_FILE = "input.json"
OUTPUT_DIR = "output"

# читаем JSON
with open(INPUT_FILE, "r", encoding="utf-8") as f:
    data = json.load(f)

if "files" not in data or not isinstance(data["files"], list):
    raise ValueError("Ожидался массив 'files' в JSON")

# создаём output/
os.makedirs(OUTPUT_DIR, exist_ok=True)

for file in data["files"]:
    # пропускаем telegram id и любые служебные записи
    if "name" not in file or "content" not in file:
        continue

    file_path = Path(OUTPUT_DIR) / file["name"]
    file_path.parent.mkdir(parents=True, exist_ok=True)

    content = file["content"]

    # если content — объект, сериализуем в JSON
    if isinstance(content, (dict, list)):
        content = json.dumps(content, indent=2, ensure_ascii=False)

    with open(file_path, "w", encoding="utf-8") as f:
        f.write(content)

    print(f"✔ created: {file['name']}")

print("\nГотово. Astro-проект собран в папке output/")

File: json_host/generate.mjs
import fs from "fs";
import path from "path";

const INPUT_FILE = "./input.json";
const OUTPUT_DIR = "./output";

const raw = fs.readFileSync(INPUT_FILE, "utf-8");
const data = JSON.parse(raw);

if (!Array.isArray(data.files)) {
  throw new Error("Ожидался массив files");
}

fs.mkdirSync(OUTPUT_DIR, { recursive: true });

for (const file of data.files) {
  if (!file.name || file.content === undefined) continue;

  const filePath = path.join(OUTPUT_DIR, file.name);
  fs.mkdirSync(path.dirname(filePath), { recursive: true });

  const content =
    typeof file.content === "object"
      ? JSON.stringify(file.content, null, 2)
      : file.content;

  fs.writeFileSync(filePath, content, "utf-8");
  console.log("✔ created:", file.name);
}

console.log("Готово");